src/lib/convenience/RegisterProviderRequest.cpp

```diff
@@ -88,11 +88,12 @@ std::string RegisterProviderRequest::render(Format format, std::string indent)
 */
 std::string RegisterProviderRequest::check
 (
-  RequestType  requestType,
-  Format       format,
-  std::string  indent,
-  std::string  predetectedError,
-  int          counter
+  ConnectionInfo* ciP,
+  RequestType     requestType,
+  Format          format,
```

Comment:
Now that ConnectionInfo is sent to the function (valid for more than one check-function in this PR), you could remove format from the signature and internally use ciP->outFormat.